apps/s_client.c

```diff
@@ -1370,6 +1376,12 @@ int s_client_main(int argc, char **argv)
         BIO_printf(bio_err, "%s: Can't use both -4 and -6\n", prog);
         goto opthelp;
     }
+    if (noservername && servername != NULL) {
+        BIO_printf(bio_err,
+                   "%s: Can't use -servername and -noservername together\n",
+                   prog);
+        goto opthelp;
+    }
```

Comment:
_Side note:_ we seem to have different schools of thought, and might need to align ourselves toward a common line of thinking...  there are other cases in this same sub-command where an option overrules a previous one rather than forbidding the combination, such as `-host`, `-port`, `-connect` and `-unix`...